### PR TITLE
feat!: Transfer Fees to Treasury

### PR DESCRIPTION
**Pull request by apopiak**
Use the `TransferFees` type introduced in https://github.com/galacticcouncil/warehouse/pull/42 in the runtimes and fix various type/integration issues.

## Related Issue

Fixes: #412

## Motivation and Context

This transfers fees to the treasury instead of burning them.
In addition it also unblocks other PRs depending on a recent version of the warehouse.

## Checklist:

- [ ] I have updated the documentation if necessary.
- [ ] I have added tests to cover my changes, regression test if fixing an issue.
- [x] This is a breaking change.

### CODE REVIEW

File: runtime/testing-basilisk/src/lib.rs [hunk 1]
```diff
@@ -304,6 +308,7 @@ impl pallet_transaction_multi_payment::Config for Runtime {
 	type WithdrawFeeForSetCurrency = MultiPaymentCurrencySetFee;
 	type WeightToFee = WeightToFee;
 	type NativeAssetId = ();
```
Comment (enthusiastmartin) [inline runtime/testing-basilisk/src/lib.rs:310]: hm.not part of this PR: but why there is a missing native asset id ? i wonder how this could work without it 🤔

**Review by enthusiastmartin — COMMENTED** [review]
it looks good to me. it is just an integration of work which hopefully has been done and tested. 

I'll leave to @Roznovjak to approve.

File: runtime/testing-basilisk/src/lib.rs (hunk 1 above)
Comment (Roznovjak) [reply runtime/testing-basilisk/src/lib.rs:310]: https://github.com/paritytech/substrate/blob/4037345b8c0491ae5b74fcb6d131b3ff304f0bed/primitives/runtime/src/traits.rs#L297

**Review by Roznovjak — APPROVED** [review]
Looks good. There is probably no documentation requiring changes. But I'm not sure about tests. Do we want to add some tests? The problem with testing is that I'm not sure how we can call a TX and charge a fee for it without using dedicated struct for handling fees (`ChargeTransactionPayment`, as used in the tests in the TX multi payment pallet).

File: runtime/testing-basilisk/src/lib.rs (hunk 1 above)
Comment (enthusiastmartin) [reply runtime/testing-basilisk/src/lib.rs:310]: well.that means we just lucky that native is 0 then. 

better to fix it.

Comment (enthusiastmartin) [reply runtime/testing-basilisk/src/lib.rs:310]: #495

File: Cargo.toml [hunk 2]
```diff
@@ -198,8 +198,8 @@ cumulus-relay-chain-interface = { git = "https://github.com/paritytech//cumulus"
 cumulus-relay-chain-local = { git = "https://github.com/paritytech//cumulus", rev = "76479e7fef3af7c8828a44647847b01afd5fefe5" }
 
 [patch."https://github.com/galacticcouncil/warehouse"]
-pallet-price-oracle = { git = "https://github.com/galacticcouncil//warehouse", rev = "c73846ec093357fea1b35da5c56ecb37592d4896"}
-pallet-relaychain-info = { git = "https://github.com/galacticcouncil//warehouse", rev = "c73846ec093357fea1b35da5c56ecb37592d4896"}
-pallet-transaction-multi-payment = { git = "https://github.com/galacticcouncil//warehouse", rev = "c73846ec093357fea1b35da5c56ecb37592d4896"}
-pallet-asset-registry= { git = "https://github.com/galacticcouncil//warehouse", rev = "c73846ec093357fea1b35da5c56ecb37592d4896"}
-hydradx-traits = { git = "https://github.com/galacticcouncil//warehouse", rev = "c73846ec093357fea1b35da5c56ecb37592d4896"}
+pallet-price-oracle = { git = "https://github.com/galacticcouncil//warehouse", rev = "50c82ab02bb5baf99b263342dc5d4cc7e62c7e88"}
```
Comment (enthusiastmartin) [inline Cargo.toml:201]: Why we dont use the very latest warehouse revision? 

this one : 7052eeab309742dccfe7e307c15fc4f780713553

Comment (enthusiastmartin) [reply Cargo.toml]: I have updated to the latest revision.
